### ISSUE TITLE
frozen on "packing ePub"

### ISSUE THREAD

**Issue by theonlygusti** [MakeYourLifeEasier/novel-ebook.com#7]
Clicking "download" downloads a 2KB ePub , the loading bar says "Almost finished, ePub is being packed", but nothing happens for 6 hours+ so I assume it's not working.

Trying to open the downloaded (2KB ePub for 300+ chapters) says "file is corrupt"

**Comment by MichaelGuardian** [comment]
Which novel did you try to download and what was the chapter range?

**Comment by theonlygusti** [comment]
@MichaelGuardian any novel, any chapter range.

I am on macOS 10.14.5, and this happened in Safari and Chrome.

**Comment by MichaelGuardian** [comment]
I can't replicate the problem on Windows. Could you create a screenshot of the console output? If I recall correctly the keyboard shortcut is Command+Option+J.

**Comment by theonlygusti** [comment]
@MichaelGuardian  there is no console output beyond this error

    [Error] Failed to load resource: the server responded with a status of 404 () (Rebirth-of-the-Thief-Who-Roamed-the-World.jpg, line 0)

which I don't believe to be significant.

---

OH WOW

I've figured it out.

I was just confused the whole time,

the "starting chapter" drop down has a "Starting Chapter" default value, which I assumed meant "use the earliest chapter". Similarly the ending chapter drop-down is auto-filled with "Ending chapter".

I was thus leaving these like that when clicking download.

Manually choosing chapters from the drop down results in me downloading a book as expected.

And earlier I think I misremembered having tested with other chapter ranges, or I probably tested with a range like "Chapter 1000" to "Ending Chapter" (which is still nonsense).

------

It might be an idea to change the interface? "Choose ending chapter..." or "Ending chapter..." instead of "Ending Chapter", or showing some error message to the user if these are unselected when Download is clicked.
